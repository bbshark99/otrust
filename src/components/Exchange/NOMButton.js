import React from 'react';

import { useExchange } from 'context/exchange/ExchangeContext';
import { useChain } from 'context/chain/ChainContext';
import { SellBtn } from './exchangeStyles';

export default function NOMButton({ onBid, onApprove }) {
  const { weakBalance, NOMallowance } = useChain();

  const { bidAmount, bidDenom, input, weak } = useExchange();

  if (bidDenom === 'strong') return <SellBtn> Input Value </SellBtn>;
  if (bidAmount.lte(weakBalance)) {
    if (input === '') return <SellBtn> Input Value </SellBtn>;
    if (NOMallowance.gte(bidAmount)) return <SellBtn onClick={onBid}>Sell {weak}</SellBtn>;
    return <SellBtn onClick={onApprove}>Approve</SellBtn>;
  }
  return <SellBtn> Low {weak} Balance </SellBtn>;
}
