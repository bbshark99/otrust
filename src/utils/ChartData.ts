import { ApolloClient, NormalizedCacheObject, gql } from '@apollo/client';
import { BigNumber } from 'bignumber.js';

import {
  ErrorCallback,
  HistoryCallback,
  IDatafeedChartApi,
  LibrarySymbolInfo,
  OnReadyCallback,
  PeriodParams,
  ResolutionString,
  ResolveCallback,
  SearchSymbolsCallback,
  SubscribeBarsCallback,
} from 'charting_library/charting_library';

const WNOM_HISTORICAL_DATA_QUERY = gql`
  query transactions($filter: String!, $skipped: Int!) {
    wnomhistoricalFrames(
      first: 1000
      skip: $skipped
      where: { type: $filter }
      orderBy: startTime
      orderDirection: asc
    ) {
      startTime
      endTime
      startPrice
      endPrice
      minPrice
      maxPrice
    }
  }
`;

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const PERIOD_MAP = {
  '1': 'Minute',
  '15': 'QuarterHour',
  '60': 'Hour',
  '480': 'QuarterDay',
  '1D': 'Day',
  '1W': 'Week',
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const NOM_SYMBOL_INFO: LibrarySymbolInfo = {
  name: 'NOM',
  full_name: 'NOM',
  description: 'NOM',
  exchange: 'EthBondCurve',
  listed_exchange: 'EthBondCurve',
  format: 'price',
  has_intraday: true,
  has_no_volume: true,
  has_seconds: false,
  has_ticks: false,
  has_daily: true,
  has_empty_bars: true,
  data_status: 'streaming',
  original_currency_code: 'ETH',
  currency_code: 'ETH',
  // @ts-ignore
  supported_resolutions: Object.keys(PERIOD_MAP),
  intraday_multipliers: Object.keys(PERIOD_MAP),
  //supported_resolutions: [],
  ticker: 'NOM',
  timezone: 'Etc/UTC',
  type: 'crypto',
  session: '24x7',
  minmov: 1,
  pricescale: 10000000000,
};

function formatPrice(price: string) {
  return new BigNumber(price).shiftedBy(-18).toFixed(10);
}

// @ts-ignore
export class ChartData implements IDatafeedChartApi {
  private apolloClient: ApolloClient<NormalizedCacheObject>;

  constructor(apolloClient: ApolloClient<NormalizedCacheObject>) {
    this.apolloClient = apolloClient;
  }

  public async onReady(cb: OnReadyCallback) {
    cb({
      exchanges: [
        {
          value: NOM_SYMBOL_INFO.exchange,
          name: NOM_SYMBOL_INFO.exchange,
          desc: NOM_SYMBOL_INFO.exchange,
        },
      ],
      symbols_types: [
        {
          value: NOM_SYMBOL_INFO.type,
          name: NOM_SYMBOL_INFO.type,
        },
      ],
      supported_resolutions: NOM_SYMBOL_INFO.supported_resolutions,
      currency_codes: ['ETH'],
      units: {},
      supports_marks: false,
      supports_timescale_marks: false,
      supports_time: true,
    });
  }

  public async searchSymbols(
    userInput: string,
    exchange: string,
    symbolType: string,
    onResult: SearchSymbolsCallback
  ) {
    onResult([
      {
        symbol: NOM_SYMBOL_INFO.ticker!,
        full_name: NOM_SYMBOL_INFO.name,
        description: NOM_SYMBOL_INFO.description,
        exchange: NOM_SYMBOL_INFO.exchange,
        ticker: NOM_SYMBOL_INFO.ticker!,
        type: NOM_SYMBOL_INFO.type,
      },
    ]);
  }

  resolveSymbol(symbolName: string, onSuccess: ResolveCallback) {
    // TradingView requires this setTimeout
    setTimeout(() => onSuccess(NOM_SYMBOL_INFO), 0);
  }

  public async getBars(
    symbolInfo: LibrarySymbolInfo,
    resolution: string,
    periodParams: PeriodParams,
    onResult: HistoryCallback,
    onError: ErrorCallback
  ) {
    try {
      const resp = await this.apolloClient.query({
        query: WNOM_HISTORICAL_DATA_QUERY,
        variables: {
          // @ts-ignore
          filter: PERIOD_MAP[resolution] || 'Hour',
          skipped: 0,
        },
      });

      const {
        data: { wnomhistoricalFrames: frames },
      } = resp;

      const bars = frames.map(
        (frame: {
          startTime: any;
          startPrice: any;
          maxPrice: any;
          minPrice: any;
          endPrice: any;
        }) => ({
          time: parseInt(frame.startTime, 10) * 1000,
          open: formatPrice(frame.startPrice),
          high: formatPrice(frame.maxPrice),
          low: formatPrice(frame.minPrice),
          close: formatPrice(frame.endPrice),
          volume: 0,
        })
      );

      console.log('bars', bars);

      if (periodParams.firstDataRequest) {
        onResult(bars, { noData: false });
      } else {
        onResult([], { noData: true });
      }
    } catch (e) {
      onError(`${e}`);
    }
  }

  public async subscribeBars(
    symbolInfo: LibrarySymbolInfo,
    resolution: ResolutionString,
    onTick: SubscribeBarsCallback,
    listenerGuid: string,
    onResetCacheNeededCallback: () => void
  ) {
    console.log('subscribeBars', {
      symbolInfo,
      resolution,
      listenerGuid,
      onResetCacheNeededCallback,
    });

    // TODO: implement data subscription
  }

  public async unsubscribeBars(listenerGuid: string) {
    console.log('unsubscribeBars', listenerGuid);
    // TODO: implement data unsubscribe
  }
}
