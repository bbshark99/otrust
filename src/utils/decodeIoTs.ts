import { isLeft, getOrElse } from 'fp-ts/lib/Either';
import * as t from 'io-ts';

export function decodeIoTs<T, O, I>(validator: t.Type<T, O, I>, input: I): T {
  const result = validator.decode(input);
  const decoded = getOrElse(() => result as unknown as T)(result);

  if (isLeft(result)) {
    throw new Error(result.left.toString());
  }

  return decoded;
}

export function encodeIoTs<T, O, I>(validator: t.Type<T, O, I>, input: T): O {
  const result = validator.encode(input);
  return result;
}
