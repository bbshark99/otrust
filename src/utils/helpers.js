export function getFirstMessage(messagesObject) {
  let filteredArray = Object.values(messagesObject).filter(message => message);
  return filteredArray.length ? filteredArray[0] : null;
}
