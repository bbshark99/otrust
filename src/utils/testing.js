import { ThemeProvider } from 'styled-components';
import ApolloClient, { InMemoryCache } from 'apollo-boost';
import { ApolloProvider } from '@apollo/client';
import { render } from '@testing-library/react';
import { BigNumber } from 'bignumber.js';

import { darkNew } from 'theme/theme';
import { ChainContext } from '../context/chain/ChainContext';
import { ExchangeContext } from '../context/exchange/ExchangeContext';
import { ModalContext } from '../context/modal/ModalContext';
import { UpdateExchangeContext } from '../context/exchange/ExchangeContext';

const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_ENDPOINT,
  cache: new InMemoryCache(),
});

export const renderWithTheme = (Component, props, children) => {
  if (children) {
    return render(
      <ThemeProvider theme={darkNew}>
        <Component {...props}>{children}</Component>
      </ThemeProvider>,
    );
  }
  return render(
    <ThemeProvider theme={darkNew}>
      <Component {...props} />
    </ThemeProvider>,
  );
};

export const ThemeWrapper = (Component, props, children) => {
  return (
    <ThemeProvider theme={darkNew}>
      <Component {...props}>{children}</Component>
    </ThemeProvider>
  );
};

export const ChainContextWrapper = (children, contextProps) => {
  return (
    <ChainContext.Provider
      value={{
        blockNumber: BigNumber(0),
        currentETHPrice: BigNumber(0),
        currentNOMPrice: BigNumber(0),
        NOMallowance: BigNumber(0),
        strongBalance: BigNumber(0),
        supplyNOM: BigNumber(0),
        weakBalance: BigNumber(0),
        theme: { darkNew },
        ...contextProps,
      }}
    >
      {children}
    </ChainContext.Provider>
  );
};

export const ExchangeContextWrapper = (children, contextProps) => {
  return (
    <ExchangeContext.Provider
      value={{
        askAmount: BigNumber(0),
        bidAmount: BigNumber(0),
        bidDenom: 'strong',
        status: 'Not Approved',
        strong: 'ETH',
        weak: 'NOM',
        ...contextProps,
      }}
    >
      {children}
    </ExchangeContext.Provider>
  );
};

export const UpdateExchangeContextWrapper = (children, contextProps) => {
  return (
    <UpdateExchangeContext.Provider
      value={{ objDispatch: () => {}, strDispatch: () => {}, setInputPending: false, ...contextProps }}
    >
      {children}
    </UpdateExchangeContext.Provider>
  );
};

export const ModalContextWrapper = (children, contextProps) => {
  return (
    <ModalContext.Provider
      value={{ handleModal: () => {}, modal: false, modalContent: 'Modal Content', ...contextProps }}
    >
      {children}
    </ModalContext.Provider>
  );
};

export const renderWithContext = (Component, props, contextValues) => {
  return render(
    <ThemeProvider theme={darkNew}>
      <ApolloProvider client={client}>
        <ChainContext.Provider
          value={{
            supplyNOM: BigNumber(0),
            blockNumber: BigNumber(0),
            currentETHPrice: BigNumber(0),
            currentNOMPrice: BigNumber(0),
            NOMallowance: BigNumber(0),
            strongBalance: BigNumber(0),
            weakBalance: BigNumber(0),
            theme: darkNew,
            ...contextValues,
          }}
        >
          <UpdateExchangeContext.Provider
            value={{ objDispatch: jest.fn(), strDispatch: jest.fn(), setInputPending: false, ...contextValues }}
          >
            <ExchangeContext.Provider
              value={{
                askAmount: BigNumber(0),
                bidAmount: BigNumber(0),
                bidDenom: 'strong',
                status: 'Not Approved',
                strong: 'ETH',
                weak: 'NOM',
                ...contextValues,
              }}
            >
              <ModalContext.Provider
                value={{ handleModal: jest.fn(), modal: false, modalContent: 'Modal Content', ...contextValues }}
              >
                <Component {...props} />
              </ModalContext.Provider>
            </ExchangeContext.Provider>
          </UpdateExchangeContext.Provider>
        </ChainContext.Provider>
      </ApolloProvider>
    </ThemeProvider>,
  );
};
