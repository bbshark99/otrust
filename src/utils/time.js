const SECONDS_IN_ONE_DAY = 86400;

export const getTimeStampDaysAgo = numberOfDays => {
  const date = new Date();
  return Math.floor(date.getTime() / 1000 - numberOfDays * SECONDS_IN_ONE_DAY).toString();
};

export const getTimeStampMonthsAgo = numberOfMonths => {
  const date = new Date();
  date.setMonth(date.getMonth() - numberOfMonths);
  return Math.floor(date.getTime() / 1000).toString();
};
