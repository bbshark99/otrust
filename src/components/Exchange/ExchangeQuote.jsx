import React, { useCallback, useEffect } from 'react';
import { BigNumber } from 'bignumber.js';
import ConfirmTransactionModal from 'components/Modals/components/ConfirmTransactionModal';
import PendingModal from 'components/Modals/components/PendingModal';
import ApproveTokensModal from 'components/Modals/components/ApproveTokensModal';
import RequestFailedModal from 'components/Modals/components/RequestFailedModal';
import TransactionCompletedModal from 'components/Modals/components/TransactionCompletedModal';
import TransactionFailedModal from 'components/Modals/components/TransactionFailedModal';
import { useWeb3React } from '@web3-react/core';

import { useChain } from 'context/chain/ChainContext';
import { BondingCont, NOMCont } from 'context/chain/contracts';
import { useExchange, useUpdateExchange } from 'context/exchange/ExchangeContext';
import {
  ExchangeItem,
  Sending,
  Receiving,
  ExchangeInput,
  MaxBtn,
  ReceivingValue,
  ExchangeButton,
} from './exchangeStyles';
import { useModal } from 'context/modal/ModalContext';
import NOMButton from 'components/Exchange/NOMButton';
import { format18, parse18 } from 'utils/math';

export default function ExchangeQuote({ strength }) {
  const { strongBalance, weakBalance } = useChain();
  const { handleModal } = useModal();
  const { library } = useWeb3React();

  const bondContract = BondingCont(library);
  const NOMcontract = NOMCont(library);

  const { askAmount, bidAmount, approveAmount, bidDenom, input, output, approve, strong, weak } = useExchange();
  const { NOMallowance } = useChain();

  useEffect(() => {
    console.log('Input: ', input);
    console.log('Output: ', output);
    console.log('Approve: ', approve);
  });

  const { objDispatch, strDispatch } = useUpdateExchange();

  const getAskAmount = useCallback(
    async (askAmountState, bidAmountUpdate, textStrength) => {
      var askAmountUpdate = askAmountState;

      switch (textStrength) {
        case 'strong':
          console.log('Strong: ', bidAmountUpdate.toFixed(0));
          askAmountUpdate = await bondContract.buyQuoteETH(bidAmountUpdate.toFixed(0));
          console.log('Pull Strong Ask Amount', askAmountUpdate);
          break;

        case 'weak':
          askAmountUpdate = await bondContract.sellQuoteNOM(bidAmountUpdate.toFixed(0));
          console.log('Pull Weak Ask Amount', askAmountUpdate);
          break;

        default:
          console.error('Denom not set');
      }
      return new BigNumber(askAmountUpdate.toString());
    },
    [bondContract],
  );

  const onApprove = async () => {
    if (weakBalance.gte(bidAmount)) {
      if (bidAmount.gt(NOMallowance)) {
        const approvalAmount = bidAmount.minus(NOMallowance);
        let objUpdate = new Map();
        objUpdate = objUpdate.set('approveAmount', approvalAmount);

        objDispatch({
          type: 'update',
          value: objUpdate,
        });

        let strUpdate = new Map();
        strUpdate = strUpdate.set('approve', format18(approvalAmount).toString());

        strDispatch({
          type: 'update',
          value: strUpdate,
        });

        handleModal(<ApproveTokensModal onConfirmApprove={onConfirmApprove} />);
      } else {
        handleModal(<ConfirmTransactionModal submitTrans={submitTrans} />);
      }
    } else {
      handleModal(<TransactionFailedModal error={`${weak} Balance too low`} />);
    }
  };

  const onConfirmApprove = () => {
    try {
      console.log('Approve Amount Before confirming: ', approve);
      handleModal(<ConfirmTransactionModal isApproving submitTrans={submitTrans} />);
    } catch (e) {
      handleModal(<TransactionFailedModal error={e.code + '\n' + e.message.slice(0, 80) + '...'} />);
    }
  };

  const submitTrans = async (isApproving, slippage, gasPrice) => {
    handleModal(<PendingModal isApproving={isApproving} />);

    if (isApproving) {
      if (!approveAmount) return;
      try {
        console.log('Approve Amount: ', approveAmount.toString());
        console.log('Approve Amount BG: ', approveAmount);
        console.log('Approve Amount; string', approve);
        console.log('Approve Amount: Bid', bidAmount.toString());

        let tx = await NOMcontract.increaseAllowance(bondContract.address, approveAmount.toFixed(8), {
          gasPrice: gasPrice.toFixed(0),
        });

        tx.wait().then(() => {
          handleModal(<TransactionCompletedModal tx={tx} />);
        });
      } catch (e) {
        console.log(e);
        handleModal(<TransactionFailedModal error={e.code + '\n' + e.message.slice(0, 80) + '...'} />);
      }
    } else {
      if (!bidAmount || !askAmount) return;
      try {
        let tx;
        let gasFee;

        switch (bidDenom) {
          case 'strong':
            // Preparing for many tokens / coins
            switch (strong) {
              case 'ETH':
                const gasFeeRaw = await bondContract.estimateGas.buyNOM(askAmount.toFixed(0), slippage.toFixed(0), {
                  value: bidAmount.toFixed(0),
                });

                gasFee = new BigNumber(gasFeeRaw.toString());

                const bidAmountUpdate = bidAmount.minus(gasFee.times(gasPrice));
                const askAmountUpdateRaw = await bondContract.buyQuoteETH(bidAmountUpdate.toFixed(0));
                const askAmountUpdate = new BigNumber(askAmountUpdateRaw.toString());

                tx = await bondContract.buyNOM(askAmountUpdate.toFixed(0), slippage.toFixed(0), {
                  value: bidAmountUpdate.toFixed(0),
                  gasPrice: gasPrice.toFixed(0),
                  gasLimit: gasFee.toFixed(0),
                });

                tx.wait().then(() => {
                  handleModal(<TransactionCompletedModal tx={tx} />);
                });

                break;
              default: {
              }
            }
            break;

          case 'weak':
            switch (weak) {
              case 'wNOM':
                tx = await bondContract.sellNOM(bidAmount.toFixed(0), askAmount.toFixed(0), slippage.toFixed(0), {
                  gasPrice: gasPrice.toFixed(0),
                });

                tx.wait().then(() => {
                  handleModal(<TransactionCompletedModal tx={tx} />);
                });
                break;
              default: {
              }
            }
            break;

          default:
            console.log();
        }
      } catch (e) {
        // eslint-disable-next-line no-console
        console.error(e.code, e.message.message);
        // alert(e.message)
        handleModal(<TransactionFailedModal error={e.code + '\n' + e.message.slice(0, 80) + '...'} />);
      }
    }
  };

  const onBid = () => {
    switch (true) {
      case bidDenom !== strength:
        handleModal(<RequestFailedModal error="Please enter amount" />);
        break;
      case strength === 'strong' && strongBalance.gte(bidAmount):
        handleModal(<ConfirmTransactionModal submitTrans={submitTrans} />);
        break;
      case strength === 'weak' && weakBalance.gte(bidAmount):
        handleModal(<ConfirmTransactionModal submitTrans={submitTrans} />);
        break;
      default:
        handleModal(<RequestFailedModal error="Insufficient funds" />);
    }
  };

  const onMax = async () => {
    let strUpdate = new Map();
    strUpdate.set('bidDenom', strength);
    let bidMaxValue = strength === 'strong' ? strongBalance : weakBalance;

    strUpdate.set('input', format18(bidMaxValue).toString());

    let askAmountUpdate;

    try {
      askAmountUpdate = await getAskAmount(askAmount, bidMaxValue, strength);
    } catch (err) {
      if (err) {
        handleModal(<RequestFailedModal error={err.error.message} />);
      }
    }

    let objUpdate = new Map();

    objUpdate = objUpdate.set('askAmount', askAmountUpdate);

    objUpdate = objUpdate.set('bidAmount', bidMaxValue);

    objDispatch({
      type: 'update',
      value: objUpdate,
    });

    strUpdate.set('output', format18(new BigNumber(askAmountUpdate.toString())).toFixed(8));

    strDispatch({
      type: 'update',
      value: strUpdate,
    });
  };

  const onTextChange = useCallback(
    async (evt, textStrength) => {
      evt.preventDefault();
      const floatRegExp = new RegExp(/(^(?=.+)(?:[1-9]\d*|0)?(?:\.\d+)?$)|(^\d\.$)/);
      console.log('Component Strength: ', strength);
      console.log('Text Strength: ', textStrength);
      console.log('Bid Denom: ', bidDenom);
      let strUpdate = new Map();
      switch (true) {
        case bidDenom === strength && input === evt.target.value.toString():
          break;
        case evt.target.value === '' || evt.target.value === '.':
          {
            let objUpdate = new Map();

            objUpdate = objUpdate.set('askAmount', new BigNumber(0));

            objUpdate = objUpdate.set('bidAmount', new BigNumber(0));

            objDispatch({
              type: 'update',
              value: objUpdate,
            });
          }

          strUpdate = strUpdate.set('bidDenom', strength);

          strUpdate = strUpdate.set('input', evt.target.value.toString());

          strUpdate = strUpdate.set('output', '');

          strDispatch({
            type: 'update',
            value: strUpdate,
          });

          break;
        case floatRegExp.test(evt.target.value.toString()):
          console.log('Input after test', evt.target.value);
          const bidAmountUpdate = parse18(new BigNumber(parseFloat(evt.target.value).toString()));

          const inputUpdate = evt.target.value.toString();

          if (bidDenom !== strength) {
            strUpdate = strUpdate.set('bidDenom', strength);
          }

          var askAmountUpdate;

          try {
            askAmountUpdate = await getAskAmount(askAmount, bidAmountUpdate, textStrength);
          } catch (err) {
            if (err) {
              console.log(err.error.message);
              handleModal(<RequestFailedModal error={err.error.message} />);
            }
            break;
          }

          let objUpdate = new Map();

          objUpdate = objUpdate.set('askAmount', new BigNumber(askAmountUpdate.toString()));

          objUpdate = objUpdate.set('bidAmount', bidAmountUpdate);

          objDispatch({
            type: 'update',
            value: objUpdate,
          });

          strUpdate = strUpdate.set('input', inputUpdate);

          strUpdate = strUpdate.set('output', format18(new BigNumber(askAmountUpdate.toString())).toFixed(8));

          strDispatch({
            type: 'update',
            value: strUpdate,
          });

          break;
        default:
          handleModal(<RequestFailedModal error="Please enter numbers only. Thank you!" />);
      }
    },
    [askAmount, bidDenom, getAskAmount, handleModal, input, objDispatch, strDispatch, strength],
  );

  return (
    <ExchangeItem>
      <strong>Bid {strength === 'strong' ? strong : weak}</strong>
      <Sending>
        <strong>I'm bidding</strong>
        <ExchangeInput
          type="text"
          onChange={evt => onTextChange(evt, strength)}
          value={bidDenom === strength ? input : ''}
          placeholder="0.00"
        />
        {strength === 'strong' ? strong : weak}
        <MaxBtn onClick={() => onMax()}>Max</MaxBtn>
      </Sending>
      <Receiving>
        <strong>I'm asking</strong>
        <ReceivingValue>
          {strength === bidDenom ? output : ''} {strength === 'strong' ? weak : strong}
        </ReceivingValue>
      </Receiving>
      {strength === 'strong' ? (
        bidDenom === 'weak' ? (
          <ExchangeButton>Input Value</ExchangeButton>
        ) : bidAmount.lte(strongBalance) ? (
          input === '' ? (
            <ExchangeButton>Input Value</ExchangeButton>
          ) : (
            <ExchangeButton onClick={onBid}>Buy {strength === 'strong' ? weak : strong}</ExchangeButton>
          )
        ) : (
          <ExchangeButton>Low {strong} Balance</ExchangeButton>
        )
      ) : (
        <NOMButton onBid={onBid} onApprove={onApprove} />
      )}
    </ExchangeItem>
  );
}
