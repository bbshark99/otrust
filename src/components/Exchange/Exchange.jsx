import React from 'react';

import { ExchangeWrapper } from './exchangeStyles';
import ExchangeQuote from './ExchangeQuote';

export default function Exchange() {
  return (
    <ExchangeWrapper>
      <div id="tour-buy">
        <ExchangeQuote strength="strong" />
      </div>
      <div id="tour-sell">
        <ExchangeQuote strength="weak" />
      </div>
    </ExchangeWrapper>
  );
}
