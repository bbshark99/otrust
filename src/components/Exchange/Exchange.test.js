import { cleanup } from '@testing-library/react';

import { renderWithContext } from 'utils/testing';
import Exchange from './Exchange';

describe('Given an Exchange component', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', async () => {
      const { asFragment } = renderWithContext(Exchange);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
