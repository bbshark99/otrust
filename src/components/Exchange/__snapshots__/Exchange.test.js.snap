// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Exchange component when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM fYAiQM"
  >
    <div
      class="sc-kHWWYL gNGEIQ"
    >
      <button
        class="sc-iqAclL gQfUwo"
        data-testid="exchanges-modals-buy-button"
      >
        Buy wNOM
      </button>
      <button
        class="sc-iqAclL sc-crzoAE gQfUwo dZRTZd"
        data-testid="exchanges-modals-sell-button"
      >
        Sell wNOM
      </button>
    </div>
    <div
      id="tour-buy"
    >
      <div
        class="sc-gtsrHT hfTecP"
      >
        <strong>
          Buy NOM
        </strong>
        <div
          class="sc-dlnjwi hyqJyT"
        >
          <div
            class="sc-eCApnc cPbuco"
            style="width: 100%; padding-right: 16px;"
          >
            <strong>
              I'm buying for
            </strong>
            <input
              class="sc-fujyAs kMRocn"
              data-testid="exchange-strong-balance-input"
              placeholder="0.00"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-eCApnc cPbuco"
          >
            ETH
            <button
              class="sc-gKAaRy kseRin"
              data-testid="max-value-button"
            >
              Max
            </button>
          </div>
        </div>
        <div
          class="sc-jSFjdj cQwxlU"
        >
          <strong>
            You will receive
          </strong>
          <span
            class="sc-pNWdM iHufyo"
            data-testid="exchange-weak-balance"
          >
             NOM
          </span>
        </div>
        <button
          class="sc-iqAclL gQfUwo"
        >
          Buy NOM
        </button>
      </div>
    </div>
    <div
      id="tour-sell"
    >
      <div
        class="sc-gtsrHT hfTecP"
      >
        <strong>
          Sell NOM
        </strong>
        <div
          class="sc-dlnjwi hyqJyT"
        >
          <div
            class="sc-eCApnc cPbuco"
            style="width: 100%; padding-right: 16px;"
          >
            <strong>
              I'm selling
            </strong>
            <input
              class="sc-fujyAs kMRocn"
              data-testid="exchange-strong-balance-input"
              placeholder="0.00"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-eCApnc cPbuco"
          >
            NOM
            <button
              class="sc-gKAaRy kseRin"
              data-testid="max-value-button"
            >
              Max
            </button>
          </div>
        </div>
        <div
          class="sc-jSFjdj cQwxlU"
        >
          <strong>
            You will receive
          </strong>
          <span
            class="sc-pNWdM iHufyo"
            data-testid="exchange-weak-balance"
          >
             ETH
          </span>
        </div>
        <button
          class="sc-iqAclL sc-crzoAE gQfUwo dZRTZd"
        >
           Input Value 
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
